src/features/Faucet/helpers.tsx

```diff
@@ -84,20 +104,37 @@ export const makeTxReceipt = (
     assets
   })!;
 
-  return {
+  const txReceipt = {
     asset: baseAsset,
     baseAsset,
     txType: ITxType.FAUCET,
     status: ITxStatus.PENDING,
     receiverAddress: txResult.to,
     amount: formatEther(txResult.value),
     data: txResult.data,
-    gasPrice: BigNumber.from(txResult.gasPrice),
     gasLimit: BigNumber.from(txResult.gasLimit),
     to: txResult.to,
     from: txResult.from,
     value: BigNumber.from(txResult.value),
     nonce: BigNumber.from(txResult.nonce),
     hash: txResult.hash
   };
+
+  if ('maxFeePerGas' in txResult && 'maxPriorityFeePerGas' in txResult) {
+    /* Type 2 tx */
+    return {
+      ...txReceipt,
+      status: ITxStatus.PENDING,
+      maxFeePerGas: BigNumber.from(txResult.maxFeePerGas),
+      maxPriorityFeePerGas: BigNumber.from(txResult.maxPriorityFeePerGas),
+      type: 2
+    };
+  } else {
+    /* Type 1 tx */
+    return {
+      ...txReceipt,
+      status: ITxStatus.PENDING,
+      gasPrice: BigNumber.from(txResult.gasPrice)
+    };
+  }
```

Comment:
Maybe use [`makeTxConfigFromTx`](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/helpers/transaction.ts#L253).

Reply:
Is there a reason for not just constructing the receipt based on txConfig? Fx. using: https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/helpers/transaction.ts#L99

Reply:
See other comment

Reply:
As soon as the tx config is created correctly shouldn't the other function work fine to adapt it to a receipt?